tools/run_tests/run_microbenchmark.py

```diff
@@ -203,11 +203,27 @@ def run_summary(bm_name, cfg, base_json_name):
 
 
 def collect_summary(bm_name, args):
-    heading('Summary: %s [no counters]' % bm_name)
-    text(run_summary(bm_name, 'opt', bm_name))
-    heading('Summary: %s [with counters]' % bm_name)
-    text(run_summary(bm_name, 'counters', bm_name))
+    # no counters, run microbenchmark and add summary
+    # both to HTML report and to console.
+    nocounters_heading = 'Summary: %s [no counters]' % bm_name
+    nocounters_summary = run_summary(bm_name, 'opt', bm_name)
+    heading(nocounters_heading)
+    text(nocounters_summary)
+    print(nocounters_heading)
+    print(nocounters_summary)
+
+    # with counters, run microbenchmark and add summary
+    # both to HTML report and to console.
+    counter_heading = 'Summary: %s [with counters]' % bm_name
+    counters_summary = run_summary(bm_name, 'counters', bm_name)
+    heading(counter_heading)
+    text(counters_summary)
+    print(counters_heading)
+    print(counters_summary)
+
     if args.bq_result_table:
+        # TODO(jtattermusch): currently there is no way to differentiate
```

Comment:
@ctiller  this actually seems problematic - AFAICT  there is no way of distinguishing the data from "counters" and "nocounters" runs in BQ.
Would it make sense to stop uploading the data for "counters" to at least know what's stored in BQ?

Reply:
So we smash them together right now... this code calls bm2bq.py, which ultimately calls:
https://github.com/grpc/grpc/blob/master/tools/profiling/microbenchmarks/bm_json.py#L181

and that code smashes together the times from the non-counters run with the counters from the counters run
and that combination is... probably what we actually want

Reply:
ok, that seems quite cryptic, so I added a TODO to add some explanatory comments in the code.